libs/types/src/v2.ts

```diff
@@ -94,4 +96,48 @@ export type AttestationElement = StateElement & {
   attestation: Attestation
 }
 
-export type V2FileContractResolutionType = unknown // TODO: replace `unknown` with the actual type
+type V2FileContractResolutionTypeBase = {
+  parent: V2FileContractElement
+}
+
+export type V2FileContractResolutionTypeExpiration =
+  V2FileContractResolutionTypeBase & {
+    type: 'expiration'
+    resolution: Record<string, never> // is this always empty for expiration type?
+  }
+export type V2FileContractResolutionTypeFinalization =
+  V2FileContractResolutionTypeBase & {
+    type: 'finalization'
+    resolution: V2FileContract
+  }
+
+export type V2FileContractResolutionTypeRenewal =
+  V2FileContractResolutionTypeBase & {
+    type: 'renewal'
+    resolution: {
+      finalRevision: V2FileContract
+      newContract: V2FileContract
+      renterRollover: Currency
+      hostRollover: Currency
+      renterSignature: Signature
+      hostSignature: Signature
+    }
+  }
+
+export type V2FileContractResolutionDataStorageProof =
+  V2FileContractResolutionTypeBase & {
+    type: 'storage proof'
```

Comment:
@n8maninger should this be camel case? wallet event types are camel case.

Reply:
I'd like to be consistent, but these are in `core`. For now we'll leave the inconsistency and discuss in a follow up if that's alright.